hooks/use-is-plugin-active/index.ts

```diff
@@ -0,0 +1,29 @@
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import type { Plugin } from '@wordpress/core-data';
+
+const ACTIVE_STATUSES: string[] = ['active', 'network-active'];
+
+/**
+ * Custom hook to check if a plugin is active and whether its resolution has finished.
+ *
+ * @param pluginName The name of the plugin to check.
+ * @returns A tuple containing two boolean values: the first indicating whether the plugin is active, 
+ * and the second indicating whether the resolution for the plugin has finished.
+ */
+export const useIsPluginActive = (pluginName: string): [boolean, boolean] => {
+	return useSelect(
+		(select) => {
+			const storeSelectors = select(coreStore);
+			const plugin: Plugin = (storeSelectors as any).getPlugin(pluginName);
```

Comment:
@fabiankaegy I've used `any` here as `getPlugin` and `hasFinishedResolution` are convenience methods which are dynamically named.

I tried using a custom interface:

```tsx
interface CoreStore {
	getPlugin: (pluginName: string) => Plugin;
	hasFinishedResolution: (storeName: string, resolution: any[]) => boolean;
}
```

but I am not sure which type to extend from so that `CoreStore` contains all selectors from `select(coreStore)`.

Reply:
Yeah, all the Data API is a bane of my existence when it comes to TS